### ISSUE TITLE
Wrong calculation of output value in GRU

### ISSUE THREAD

**Issue by jpcf** [keras-team/keras#14649]
In file /keras/layers/recurrent.py, at commit  07bb88c, the output of the GRU layer is wrongly calculated. In fact, in line 1921 we can read:
    `h = z * h_tm1 + (1 - z) * hh`
which is incorrect, since the candidate hidden state `hh` multiplies with z and not with (1-z). I guess the reason it works is because, in the end, the algorithm adjusted the weights `Wz` and `Uz` that produces z so take this into account, but I feat that the accuracy might not be achieving its full potential because of this. From the GRU original paper [https://arxiv.org/abs/1412.3555](url), on page 4, we confirm that `z` actually multiplies with `hh`, and  not the other way around:

<img width="426" alt="2021-05-16 21_31_03-1412 3555 pdf" src="https://user-images.githubusercontent.com/9438498/118410103-15a9e880-b68e-11eb-865c-2c5e85ef6167.png">

Thanks in advance!

**Comment by River861** [comment]
I agree with you :)

**Comment by sushreebarsa** [comment]
@jpcf  The links provided in the issue seem to be not working. Could you please have a look at this updated [doc](https://keras.io/api/layers/recurrent_layers/gru/) and let me know if it helps?
Thank you!
